src/IndicatorsDash.jsx

```diff
@@ -91,8 +104,17 @@ class IndicatorsDash extends React.Component {
     this.setState({ [state]: false });
   };
 
+  /**
+   * Re-Load indicators list by selecting one bioma option
+   *
+   * @param {String} state state value that controls the modal you want to close
+   */
   handleChange = (selectedOption) => {
-    this.setState({ selectedOption });
+    const { activeBlock } = this.state;
+    this.setState(
+      { selectedOption },
+      () => this.loadIndicators(activeBlock.id, selectedOption ? selectedOption.value : undefined),
```

Comment:
Why did you use this for of "setState" ? I think the call to "loadIndicators" doesn't need to wait until the state is updated (loadIndicators doesn't use "selectedOption" state).